mum/src/notifications.rs

```diff
@@ -1,25 +1,34 @@
-#[cfg(feature = "notifications")]
-use log::*;
-
+/// Try to initialize the notification daemon.
+///
+/// Does nothing if compiled without notification support.
 pub fn init() {
     #[cfg(feature = "notifications")]
     if let Err(e) = libnotify::init("mumd") {
-        warn!("Unable to initialize notifications: {}", e);
+        log::warn!("Unable to initialize notifications: {}", e);
     }
 }
 
 #[cfg(feature = "notifications")]
-pub fn send(msg: String) -> Option<bool> {
-    match libnotify::Notification::new("mumd", Some(msg.as_str()), None).show() {
-        Ok(_) => Some(true),
-        Err(e) => {
-            warn!("Unable to send notification: {}", e);
-            Some(false)
+/// Send a notification non-blocking, returning a JoinHandle that resolves to
+/// whether the notification was sent or not.
+///
+/// The notification is sent on its own thread since a timeout otherwise might
+/// block for several seconds.
+///
+/// None is returned iff the program was compiled without notification support.
+pub fn send(msg: String) -> Option<std::thread::JoinHandle<bool>> {
```

Comment:
Why do we return the join handle if we don't use it where this method is called?

Reply:
While we don't use it, I opted to leave it in in case we do want to use it in the future. It costs us nothing to drop the handle at the call site, and it also becomes an easy way to check if the send was tried at all (`send("content".to_string().is_some()`).

Reply:
Aight sure.